app/models/api/v1/report.rb

```diff
@@ -0,0 +1,226 @@
+    }
+  end
+
+  # Helpers that provide class description:
+
+  def class_json
+    clazz = @offering.clazz
+    section = clazz.section && clazz.section.length > 0 ? clazz.section : nil
+    section = section ? " (#{section})" : ""
+    {
+      name: clazz.name + section,
+      students: clazz.students.includes(:user).map { |s| student_json(s) }.sort_by { |s| s[:name] }
+    }
+  end
+
+  def student_json(student)
+    started_offering = student.report_learners.where(offering_id: @offering.id).count > 0
+    {
+      id: student.id,
+      name: student.name,
+      started_offering:  started_offering
+    }
+  end
+
+  # Helpers that provide student answers grouped by embeddable key:
+
+  def get_student_answers
+    # Collect all the student answers for given offering.
+    answers = Report::Learner.where(offering_id: @offering.id).map do |report_learner|
+      student_id = report_learner.student_id
+      answers = []
+      report_learner.answers.map do |embeddable_key, answer|
+        # Process some answers type to provide cleaner format, names, etc.
+        question_type = embeddable_type(embeddable_key)
+        answer[:type] = question_type
+        answer[:embeddable_key] = embeddable_key
+        answer[:student_id] = student_id
+        if question_type == 'Embeddable::MultipleChoice'
+          process_multiple_choice_answer(answer)
+        elsif question_type == 'Embeddable::ImageQuestion'
+          process_image_question_answer(answer)
+        end
+        answers.push(answer)
+      end
+      answers
+    end
+    # Flatten answers and group them by the embeddable key.
+    answers.flatten.sort_by { |s| s[:student_name] }.group_by { |a| a[:embeddable_key] }
+  end
+
+  def provide_no_answer_entries(answers, students_json)
+    # Provide "no answer" entries for students who started activity, but didn't respond to given question.
+    default_answer_entries = students_json.select { |s| s[:started_offering] }.map do |s|
+      {
+        student_id: s[:id],
+        answer: nil,
+        type: 'NoAnswer'
+      }
+    end
+    answers.each do |embeddable_key, embeddable_answers|
+      student_found = {}
+      embeddable_answers.each { |answer| student_found[answer[:student_id]] = true }
+      default_answer_entries.each do |default_answer|
+        unless student_found.fetch(default_answer[:student_id], false)
+          ans = default_answer.clone
+          ans[:embeddable_key] = embeddable_key
+          answers[embeddable_key].push(ans)
+        end
+      end
+    end
+  end
+
+  def process_multiple_choice_answer(hash)
+    # Make naming more consistent, otherwise we would have crazy sequence of [:answers][:answer][:answer] keys.
+    hash[:answer] = hash[:answer].map do |a|
+      {
+        choice: a[:answer],
+        is_correct: a[:correct]
+      }
+    end
+  end
+
+  def process_image_question_answer(hash)
+    # Filter out useless Blob description, provide image URL instead (and note / annotation).
```

Comment:
I feel like we should eventually push blob reformatting deeper into the reporting stack.  I was doing something similar with the feedback work.  I might come back to look at this.